### PR TITLE
feat: add recovery codes to user repository

### PR DESCRIPTION
**Pull request by grvijayan**
# Which Problems Are Solved

Adds a repository implementation to add/remove recovery codes to the users relational table.

# How the Problems Are Solved

This is achieved by:
* adding the following columns to the users table: `recovery_codes`, `recovery_code_last_successful_check`, `recovery_code_failed_attempts`
* adding the `RecoveryCodes` field to the `HumanUser` domain with fields to set recovery `codes`, `lastSuccessfullyCheckedAt` timestamp, and `failedAttempts`
* setting `recoveryCodes` in the `Get` user query statement to return a json object with details related to the recovery codes
* adding the repository-layer implementation to add/remove recovery codes and set fields related to recovery code checks.
* adding projection reducers to handle the following events: `HumanRecoveryCodesAddedEvent`, `HumanRecoveryCodesRemovedEvent`, `HumanRecoveryCodeCheckSucceededEvent`, and `HumanRecoveryCodeCheckFailedEvent`
* adding unit tests

# Additional Changes
* fix the error message when the recovery is empty in `internal/command/user_human_recovery_codes.go`
* add a new error message for empty recovery code during checks in `en.yaml`

# Additional Context
- Closes https://github.com/zitadel/zitadel/issues/11212
- Follow-up: integration tests for the reducers will be added in a different PR after this [PR](https://github.com/zitadel/zitadel/pull/11478) is merged

### CODE REVIEW

File: backend/v3/storage/database/repository/user_test.go [hunk 1]
```diff
@@ -3709,6 +3709,158 @@ func Test_user_Update(t *testing.T) {
...
+					require.NoError(t, err)
+					u.Human.RecoveryCodes = &domain.HumanRecoveryCodes{
+						Codes: []string{"code1", "code2"},
+					}
+					return u
+				}(),
+			},
+		},
+		{
+			name: "human user - append recovery codes",
```
Comment (adlerhurst) [inline backend/v3/storage/database/repository/user_test.go:3732]: can we add a test which checks that the duplicate elimination works?
